src/index.js

```diff
@@ -19,9 +19,17 @@ export default class Wallet extends EventEmitter {
     this._useInjectedInterface = false;
   }
 
+  _validSolanaInjecter = () => {
+    return (
+      window.solana &&
+      window.solana.providerUrl &&
+      window.solana.providerUrl.origin === this._providerUrl.origin
```

Comment:
Why do we need to match the injected URL with the wallet URL here? Injected providers don't need a provider URL. Shouldn't we remove/ignore or force-emtpy the provider URL in the constructor, so that all injected wallet can have the same interface.

Like in serum dex, there are only 2 wallets: **solet.io** and **injected**, where the second wallet allow any browser extension to polyfill.

Reply:
If only match the injected object window.solana exist, the injected wallet will take control even if the user's selection is a web wallet, and will block the web wallet.

Based on this change, injected wallet does not need to update any code in sol-wallet-adapter. But injected wallet needs to add object in their own injected solana object.

More details can be found:
http://blog.mathwallet.org/?p=2431

Reply:
> If only match the injected object window.solana exist, the injected wallet will take control even if the user's selection is a web wallet, and will block the web wallet.

Lets look at this case:
* Both MathWallet and ezDeFi inject to window.solana, somehow ezDeFi get prioritized and MathWallet never get injected.
* Dapp UI shows 3 wallets: solet.io, mathwallet, and ezdefi.
* User selects MathWallet, but solet.io is used instead.
Is this expected behavior?